dipy/tracking/fbcmeasures.pyx

```diff
@@ -0,0 +1,334 @@
+        streamlines.
+
+        Parameters
+        ----------
+        py_streamlines : list
+            A collection of streamlines, each n by 3, with n being the number of
+            nodes in the fiber.
+        kernel : Kernel object
+            A diffusion kernel object created from EnhancementKernel.
+        num_threads : int
+            Number of threads to use for OpenMP.
+        """
+        cdef:
+            int numberOfFibers
+            int maxLength
+            int dim
+            double [:, :, :] streamlines
+            int [:] streamlines_length
+            double [:, :, :] streamlines_tangent
+            int [:, :] streamlines_nearestp
+            double [:, :] streamline_scores
+            double [:] tangent
+            int lineId, pointId
+            int lineId2, pointId2
+            double score
+            int xd, yd, zd
+            double [:, :, :, :, ::1] lut
+            int N
+            int hn
+            int threads_to_use = -1
+            int all_cores = openmp.omp_get_num_procs()
+
+            double [:] score_mp
+            int [:] xd_mp
+            int [:] yd_mp
+            int [:] zd_mp
+
+        if num_threads is not None:
+            threads_to_use = num_threads
+        else:
+            threads_to_use = all_cores
+
+        if have_openmp:
+            openmp.omp_set_dynamic(0)
+            openmp.omp_set_num_threads(threads_to_use)
+        
+        # if the fibers are too short FBC measures cannot be applied, 
+        # remove these.
+        streamlines_length = np.array([len(x) for x in py_streamlines], 
+                                dtype=np.int32)
+        minLength = min(streamlines_length)
+        if minLength < 10:
+            print("The minimum fiber length is 10 points. \
+                    Shorter fibers were found and removed.")
+            py_streamlines = [x for x in py_streamlines if len(x) >= 10]
+            streamlines_length = np.array([len(x) for x in py_streamlines], 
+                                    dtype=np.int32)
+            minLength = min(streamlines_length)
+        numberOfFibers = len(py_streamlines)
+        self.streamline_length = streamlines_length
+        maxLength = max(streamlines_length)
+        
+        dim = 3
+
+        # get lookup table info
+        lut = kernel.get_lookup_table()
+        N = lut.shape[2]
+        hn = (N-1) / 2
+        
+        # prepare numpy arrays for speed
+        streamlines = np.zeros((numberOfFibers, maxLength, dim), 
+                                dtype=np.float64) * np.nan
+        streamlines_tangents = np.zeros((numberOfFibers, maxLength, dim), 
+                                dtype=np.float64)
+        streamlines_nearestp = np.zeros((numberOfFibers, maxLength), 
+                                dtype=np.int32)
+        streamline_scores = np.zeros((numberOfFibers, maxLength), 
+                                dtype=np.float64) * np.nan
```

Comment:
Not completely sure, but I think these arrays are already taking advantage of the Cython speedup because all those variables were defined with cdef (see all the cdef declaration in the top of the function). I checked them with ```cython -a fbcmeasures.pyx``` and they appeared OK.